src/lightning/pytorch/trainer/connectors/data_connector.py

```diff
@@ -420,28 +420,17 @@ def _check_dataloader_iterable(
         )
 
 
-def _worker_check(trainer: "pl.Trainer", using_spawn: bool, dataloader: object, name: str) -> None:
+def _worker_check(trainer: "pl.Trainer", dataloader: object, name: str) -> None:
     if not isinstance(dataloader, DataLoader):
         return
 
     upper_bound = suggested_max_num_workers(trainer.num_devices)
+    using_spawn = mp.get_start_method() == "spawn"
```

Comment:
Since spawn is way more expensive than fork, this is why you notice the overhead so much and `persistent_workers=True` can help. 

On Linux, default is fork and on MacOs and Windows it is spawn. So these two platforms generally benefit from the persistent_workers settings while on Linux it is not noticeable. However, once the start method is set, you can't change it, meaning if we launch multi-GPU with ddp-spawn, even on Linux the dataloaders will use spawn. That's why the warning was initially tied to ddp-spawn but it is really just a consequence of the way `torch.multiprocessing` configures itself.

There is so much confusing and noisy information out there in forums etc, when it is really quite simple.